src/post/post.service.integration.test.ts

```diff
@@ -0,0 +1,293 @@
+
+            expect(isError(result)).toBe(false);
+
+            if (isError(result)) {
+                throw new Error('Result should not be an error');
+            }
+
+            const post = getValue(result);
+            expect(post).toBeInstanceOf(Post);
+            expect(post.author.id).toBe(account.id);
+            expect(post.content).toContain(content);
+            expect(post.type).toBe(0); // PostType.Note
+            expect(post.inReplyTo).toBeNull();
+
+            // Verify the post was saved to database
+            const savedPost = await postRepository.getById(post.id!);
+            expect(savedPost).not.toBeNull();
+            expect(savedPost?.id).toBe(post.id);
+            expect(savedPost?.content).toContain(content);
+        });
+
+        it('should handle image URLs correctly', async () => {
+            const content = 'This is a test note with an image';
+            const imageUrl = new URL('https://example.com/image.jpg');
+
+            const result = await postService.createNote(
+                account,
+                content,
+                imageUrl,
+            );
+
+            expect(isError(result)).toBe(false);
+
+            if (isError(result)) {
+                throw new Error('Result should not be an error');
+            }
+
+            const post = getValue(result);
+            expect(post.imageUrl).toEqual(imageUrl);
+        });
+
+        it('should return error when image verification fails', async () => {
+            // Create a proper mock for failing GCPStorageService
+            const failingStorageService = {
+                logger: {} as unknown,
+                storage: {} as unknown,
+                bucket: {} as unknown,
+                bucketName: 'test-bucket',
+                uploadBase64Image: vi.fn(),
+                verifyImageUrl: vi
+                    .fn()
+                    .mockResolvedValue(
+                        createError(
+                            'invalid-image-url' as ImageVerificationError,
+                        ),
+                    ),
+                getPublicUrl: vi.fn(),
+                downloadImage: vi.fn(),
+                deleteFilesByPrefix: vi.fn(),
+                listFiles: vi.fn(),
+            } as unknown as GCPStorageService;
+
+            const serviceWithFailingStorage = new PostService(
+                postRepository,
+                accountService,
+                fedifyContextFactory,
+                failingStorageService,
+                moderationService,
+            );
+
+            const content = 'This is a test note';
+            const imageUrl = new URL('https://example.com/bad-image.jpg');
+
+            const result = await serviceWithFailingStorage.createNote(
+                account,
+                content,
+                imageUrl,
+            );
+
+            expect(isError(result)).toBe(true);
+            expect(getError(result as Err<string>)).toBe('invalid-image-url');
```

Comment:
How come we have to `as Err<string>` here?

Reply:
`getError` doesn't accept a `Result` it only accepts an `Error` (aliased to `Err` here)

Usually they're wrapped so like:
```
if (isError(result)) {
  getError(result); // This works because `isError` tells typescript the type
}
```

But here the `isError(result)` call isn't wrapping this we we have to cast it explicitly